### PR TITLE
feat(ci): Add sync-source-mirrors job after successful x86 build

### PR DESCRIPTION
**Pull request by MatthewWEdwards**
## Prerequisites

- [x] I have read the [Contributing Guidelines](../blob/main/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](../blob/main/CODE_OF_CONDUCT.md).
- [x] If this is a non-trivial change, I have already opened an accompanying Issue.
- [x] If applicable, I have included documentation updates alongside my code changes.

## Description

Add downloaded tarballs to the shared `source_mirrors` filesystem on
self-hosted runners. Also add BB_GENERATE_MIRROR_TARBALLS = "1" to
bitbake configuration by default for all meta-layers so that tarballs
are generated.

Some parts of the terragraph-image-x86.yml workflow are reformatted, and the file is renamed to remove the word "build"

## Test Plan

1. Delete facebook/openr GitHub tarball from source mirrors
2. Clear openr build artifacts from sstate-cache
3. Run workflow: https://github.com/terragraph/meta-terragraph/actions/runs/2794158940
4. See new openr tarball in source_mirrors: 

```
ubuntu@ip-10-0-20-186:~/source_mirrors$ ls -lh | grep openr -A 2 -B 2
-rw-r--r-- 1 ubuntu ubuntu  6.7M May 10 18:49 git2_github.com.facebook.fbzmq.git.tar.gz
-rw-r--r-- 1 ubuntu ubuntu   52M May 10 18:49 git2_github.com.facebook.folly.git.tar.gz
-rw-rw-r-- 1 ubuntu ubuntu   30M Aug  4 05:18 git2_github.com.facebook.openr.git.tar.gz
-rw-r--r-- 1 ubuntu ubuntu   21M May 10 18:49 git2_github.com.facebook.proxygen.git.tar.gz
-rw-r--r-- 1 ubuntu ubuntu  6.9M May 10 18:49 git2_github.com.facebook.wangle.git.tar.gz
```

### CODE REVIEW

File: .github/workflows/terragraph-image-x86.yml [hunk 1]
```diff
@@ -81,3 +97,20 @@ jobs:
...
+        name: downloads
+        path: /tmp/downloads
+    - name: Sync Mirrors
+      run: >-
+        find /tmp/downloads/
+        -regex ".*.\(tar\|tgz\|zip\).*"
+        -not -path "*uninative/*"
+        -and -not -path "*git2/*"
+        -not -name "*.done"
+        -exec cp {} $SELF_HOSTED_RUNNER_HOME/source_mirrors \;
```
Comment (elludraon) [inline .github/workflows/terragraph-image-x86.yml:116]: "cp -v" ?

**Review by elludraon — CHANGES_REQUESTED** [review]
Is there a reason you're enabling `BB_GENERATE_MIRROR_TARBALLS` by default instead of only in this workflow? For regular user builds the mirror tarballs would just waste disk space. If you do choose to keep it like this then `BB_GENERATE_MIRROR_TARBALLS` should also be deleted from `facebook/conf/allow-all-networks.conf`.

File: .github/workflows/terragraph-image-x86.yml [hunk 2]
```diff
@@ -1,41 +1,57 @@
...
+      run: |
+        source tg-init-build-env meta-x86 build-x86
+        USER=meta bitbake terragraph-image-x86
+    - name: Upload Rootfs
       uses: actions/upload-artifact@v3
       with:
         name: rootfsx86
         path: build-x86/tmp/deploy/images/tgx86/terragraph-image-x86-tgx86.tar.gz
         retention-days: 3
+    - name: Upload Downloads
```
[the remark this answers is not in the document]
Comment (MatthewWEdwards) [reply .github/workflows/terragraph-image-x86.yml:49]: Changed "downloads" -> "sources" because they're not all tarballs.

**Comment by MatthewWEdwards** [comment]
- Add new configuration file setting `BB_GENERATE_MIRROR_TARBALLS` to `1`
- Rename "downloads" artifact to "sources"
- Add `-v` flag to `cp`

**Review by elludraon — APPROVED** [review]

**Pull request merged**